### PR TITLE
A-009: timer for reveal

### PR DESCRIPTION
**Pull request by letier3110**
https://www.notion.so/madfissolutions/A-009-User-password-could-be-disclosed-for-unlimited-time-e916e588d3b84211b5a5ad03302d734a

added autohide for password input and for reveal view

### CODE REVIEW

File: src/components/styled-password-input/styled-password-input.tsx [hunk 1]
```diff
@@ -13,6 +13,20 @@ export const StyledPasswordInput: FC<StyledPasswordInputProps> = ({ value, ...pr
...
+      const timer = setTimeout(() => {
+        setIsSecureTextEntry(true);
+      }, 10_000);
+
+      return () => {
+        clearTimeout(timer);
+      };
+    }
+
+    return undefined;
```
Comment (LendiHop) [inline src/components/styled-password-input/styled-password-input.tsx:27]: why?

File: src/modals/reveal-private-key-modal/reveal-private-key-view/reveal-private-key-view.tsx [hunk 2]
```diff
@@ -20,6 +21,20 @@ export const RevealPrivateKeyView: FC<Props> = ({ publicKeyHash }) => {
...
+      const timer = setTimeout(() => {
+        setSecretKey(undefined);
+      }, 3 * 60_000);
+
+      return () => {
+        clearTimeout(timer);
+      };
+    }
+
+    return undefined;
```
Comment (LendiHop) [inline src/modals/reveal-private-key-modal/reveal-private-key-view/reveal-private-key-view.tsx:35]: why?

**Review by LendiHop — APPROVED** [review]

File: src/modals/reveal-private-key-modal/reveal-private-key-view/reveal-private-key-view.tsx [hunk 3]
```diff
@@ -20,6 +21,20 @@ export const RevealPrivateKeyView: FC<Props> = ({ publicKeyHash }) => {
...
+  }, [secretKey, setSecretKey]);
+
```
Comment (unixvb) [inline src/modals/reveal-private-key-modal/reveal-private-key-view/reveal-private-key-view.tsx:37]: why did you create this `useEffect` if there is already exists a timer that hides the secret key?
```
activeTimer.current = setTimeout(() => setSecretKey(undefined), OVERLAY_SHOW_TIMEOUT);
```

File: src/components/styled-password-input/styled-password-input.tsx [hunk 4]
```diff
@@ -10,17 +11,28 @@ import { useStyledPasswordInputStyles } from './styled-password-input.styles';
...
+  const { activeTimer, clearActiveTimer } = useActiveTimer();
 
   const styles = useStyledPasswordInputStyles();
 
+  const handleSecureEntryPress = () => {
+    if (isSecureTextEntry) {
+      clearActiveTimer();
+      activeTimer.current = setTimeout(() => {
+        setIsSecureTextEntry(true);
+      }, 10_000);
```
Comment (unixvb) [inline src/components/styled-password-input/styled-password-input.tsx:23]: reuse `OVERLAY_SHOW_TIMEOUT` variable and make short return inside `setTimeout` callback please

**Review by unixvb — APPROVED** [review]

**Pull request merged**